examples/AvroBlogExamples/Program.cs

```diff
@@ -53,7 +54,9 @@ static void ProduceGeneric(string bootstrapServers, string schemaRegistryUrl)
                 record.Add("IP", "127.0.0.1");
                 record.Add("Message", "a test log message");
                 record.Add("Severity", new GenericEnum(logLevelSchema, "Error"));
-                producer.ProduceAsync("log-messages", new Message<Null, GenericRecord> { Value = record })
+                await producer
+                    .ProduceAsync(Serializers.Null, valueSerializer, "log-messages",

```

Comment:
there is only one compromise left that I don't like making - that the serializers need to be specified explicitly in the avro case. alternatively, the serde registration could be made more general on `Producer`/`Consumer` (`object` rather than `Serializer<T>`), and this could be avoided. that might be better actually. another alternative is to us inheritance and have `AvroProducer` / `AvroConsumer`, but that is much worse.

Reply:
i just thought of a solution - there could be a global dictionary defined in the Confluent.Kafka.Avro assembly, with key as the producer / consumer and value a dictionary of type -> avro serdes. it would have `internal`, not `public` visibility.

Reply:
(all the above is now out of date - using inheritance).